### DISCUSSION TITLE
onSwiper in thumbnail error when using React Typescript

### DISCUSSION THREAD

**Discussion by LinhNguyen1059** [nolimits4web/swiper#5033] [Q&A]
I'm just working with swiper, I'm using typesript but got an typescript error when using `onSwiper` for `Swiper` thumbnail.
Bellow is my typescript issue show.

```
Type 'Dispatch<SetStateAction<null>>' is not assignable to type '(swiper: Swiper) => void'.
  Types of parameters 'value' and 'swiper' are incompatible.
    Type 'Swiper' is not assignable to type 'SetStateAction<null>'.
      Type 'Swiper' provides no match for the signature '(prevState: null): null'.
```

Here is my code in React:
```
const [thumbsSwiper, setThumbsSwiper] = useState(null);

<Swiper thumbs={{ swiper: thumbsSwiper }}>
  ...
</Swiper>
<Swiper
  onSwiper={setThumbsSwiper} // error here
  slidesPerView={5}
  freeMode={true}
  watchSlidesProgress={true}
>
  ...
</Swiper>
```

How can I define a type for state or fix that?
Thank you :))

**Comment by DmitryShcheklein** [comment]
Fkn 30k stars library without ts example 🙄

I fix it:
`import SwiperClass from 'swiper/types/swiper-class';`
`const [thumbsSwiper, setThumbsSwiper] = useState<SwiperClass>();`

**Reply by judewang** [reply]
@DmitrySheklein Thank you!! You saved my day!!

**Reply by GuruSistemas** [reply]
this did the trick,. altought I had to modify the import to 
import { Swiper, SwiperSlide, SwiperClass } from 'swiper/react';

Thank you so much

**Comment by antokhio** [comment]
Hi, proper way to do that:

```
const swiper = useRef<SwiperClass>(null!);
<Swiper onSwiper={(ref) => { swiper.current = ref; }} />
 ```
